File: src/commands/bootstrap.ts
import { mkdir } from "node:fs/promises";
import path from "node:path";
import { resolveConfig } from "../config.js";
import { runAgentTask } from "../agent-run.js";
import { buildBootstrapPrompt } from "../prompts-tasks.js";
import { DEFAULT_SPEC_REL } from "../project.js";

export async function runBootstrap(
  configPath: string | undefined,
  verbose: boolean,
): Promise<void> {
  const config = await resolveConfig({ configPath, requireSpec: false });

  if (config.hasSpec) {
    console.log(
      `[curious] ${DEFAULT_SPEC_REL} already exists — bootstrap will overwrite/refine it`,
    );
  } else {
    await mkdir(path.dirname(config.specPath), { recursive: true });
    console.log(`[curious] will create ${DEFAULT_SPEC_REL}`);
  }

  const prompt = await buildBootstrapPrompt(config);
  const result = await runAgentTask(config, prompt, {
    verbose,
    label: "bootstrap",
  });

  if (result.status !== "finished") {
    process.exit(1);
  }

  console.log("\n[curious] Next steps:");
  console.log("  1. Review and refine spec/SPEC.md");
  console.log("  2. curious roadmap     # expand tasks if needed");
  console.log("  3. curious run --cycle # developer → reviewer → sync");
}

File: python/src/curious/commands/bootstrap.py
from __future__ import annotations

import sys
from pathlib import Path

from curious.config import resolve_config
from curious.harness import run_harness
from curious.workspace import prepare_agent_workspace
from curious.project import DEFAULT_SPEC_REL
from curious.prompts_tasks import build_bootstrap_prompt


def run_bootstrap(config_path: str | None, verbose: bool) -> None:
    config = resolve_config(config_path=config_path, require_spec=False)
    prepare_agent_workspace(config)
    spec = Path(config.spec_path)

    if config.has_spec:
        print(f"[curious] {DEFAULT_SPEC_REL} exists — bootstrap will refine it")
    else:
        spec.parent.mkdir(parents=True, exist_ok=True)
        print(f"[curious] will create {DEFAULT_SPEC_REL}")

    prompt = build_bootstrap_prompt(config)
    result = run_harness(
        prompt,
        Path(config.cwd),
        config.llm,
        config.harness,
        verbose=verbose,
    )

    if result.status != "finished":
        print(f"[curious] bootstrap failed: {result.error}")
        sys.exit(1)

    print("\n[curious] Next steps:")
    print("  1. Review spec/SPEC.md")
    print("  2. curious-py roadmap")
    print("  3. curious-py run --cycle")
